File: bisection.cpp
#include <iostream>
#include <iomanip>

using namespace std;

double pow(double n, int p) {

	double res = n;

	for (int i = 0; i < p - 1; i++) {
		res *= n;
	}

	return res;
}

double sqrt(double n, int p) {
	double measure = n;
	double left = 0;
	double right = n;

	if (n < 1) right = 1;

	for (int i = 0; i < 200; i++) {
	
		double mid = (right + left) / 2;
		double approx = pow(mid, p);

		if (approx > measure) {
			right = mid;
		} else {
			left = mid;
		}
		
	}

	return right;
}

int main() {
	
	double a;
	int n;
	cin >> a;
	cin >> n;
		
	cout << setprecision(17);
	cout << sqrt(a, n);

	return 0;

}

File: bisection.py
a = float(input())
n = int(input())


def sqrt(n, p):
    measure = n
    left = 0
    right = n

    if n < 1: right = 1
    
    eps = 1e-10

    i = 0
    while right - left > eps:
        mid = (right + left) / 2
        approx = pow(mid, p)
        
        if approx > measure:
            right = mid
        else:
            left = mid
        
        i += 1
    
    print(i)
    return right


print(sqrt(a, n))
